src/test/java/com/puppycrawl/tools/checkstyle/gui/MainFrameModelTest.java

```diff
@@ -0,0 +1,156 @@
+// Copyright (C) 2001-2017 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.gui;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
+import java.util.Locale;
+
+import org.junit.Test;
+
+import com.puppycrawl.tools.checkstyle.api.CheckstyleException;
+import com.puppycrawl.tools.checkstyle.gui.MainFrameModel.ParseMode;
+
+public class MainFrameModelTest {
+
+    private static final String FILE_NAME_TEST_DATA = "InputJavadocAttributesAndMethods.java";
+    private static final String FILE_NAME_NON_JAVA = "NotJavaFile.notjava";
+    private static final String FILE_NAME_NON_EXISTENT = "non-existent.file";
+    private static final String FILE_NAME_NON_COMPIABLE = "InputIncorrectClass.java";
+
+    private static String getPath(String filename) {
+        return "src/test/resources/com/puppycrawl/tools/checkstyle/gui/" + filename;
+    }
+
+    @Test
+    public void testParseModeEnum() {
+        assertEquals("Plain Java", ParseMode.PLAIN_JAVA.toString());
+        assertEquals("Java with comments", ParseMode.JAVA_WITH_COMMENTS.toString());
+        assertEquals("Java with comments and Javadocs",
+                ParseMode.JAVA_WITH_JAVADOC_AND_COMMENTS.toString());
+    }
+
+    @Test
+    public void testShouldAcceptFile() {
+        final File dir = new File(getPath(""));
+        assertTrue(MainFrameModel.shouldAcceptFile(dir));
+
+        final File javaFile = new File(getPath(FILE_NAME_TEST_DATA));
+        assertTrue(MainFrameModel.shouldAcceptFile(javaFile));
+
+        final File notJavaFile = new File(getPath(FILE_NAME_NON_JAVA));
+        assertFalse(MainFrameModel.shouldAcceptFile(notJavaFile));
+
+        final File nonExistentFile = new File(getPath(FILE_NAME_NON_EXISTENT));
+        assertFalse(MainFrameModel.shouldAcceptFile(nonExistentFile));
+    }
+
+    @Test
+    public void testOpenFile() throws CheckstyleException {
```

Comment:
This test is pretty big and not easy to follow. I will try to look at it more but there is no way to split/trim it so it is easier?